securedrop_export/main.py

```diff
@@ -1,46 +1,180 @@
-        submission.exit_gracefully(ExportStatus.ERROR_METADATA_PARSING.value)
-
-    if not submission.archive_metadata.is_valid():
-        submission.exit_gracefully(ExportStatus.ERROR_ARCHIVE_METADATA.value)
-
-    if submission.archive_metadata.export_method == "start-vm":
-        submission.exit_gracefully("")
-
-    if submission.archive_metadata.export_method == "usb-test":
-        action = USBTestAction(submission)
-    elif submission.archive_metadata.export_method == "disk":
-        action = DiskExportAction(submission)
-    elif submission.archive_metadata.export_method == "disk-test":
-        action = DiskTestAction(submission)
-    elif submission.archive_metadata.export_method == "printer-preflight":
-        action = PrintPreflightAction(submission)
-    elif submission.archive_metadata.export_method == "printer":
-        action = PrintExportAction(submission)
-    elif submission.archive_metadata.export_method == "printer-test":
-        action = PrintTestPageAction(submission)
-
-    action.run()
+        safe_mkdir(DEFAULT_HOME)
+        safe_mkdir(DEFAULT_HOME, LOG_DIR_NAME)
+
+        log_file = os.path.join(DEFAULT_HOME, LOG_DIR_NAME, EXPORT_LOG_FILENAME)
+
+        # set logging format
+        log_fmt = (
+            "%(asctime)s - %(name)s:%(lineno)d(%(funcName)s) "
+            "%(levelname)s: %(message)s"
+        )
+        formatter = logging.Formatter(log_fmt)
+
+        handler = TimedRotatingFileHandler(log_file)
+        handler.setFormatter(formatter)
+
+        # For rsyslog handler
+        if platform.system() != "Linux":  # pragma: no cover
+            syslog_file = "/var/run/syslog"
+        else:
+            syslog_file = "/dev/log"
+
+        sysloghandler = SysLogHandler(address=syslog_file)
+        sysloghandler.setFormatter(formatter)
+        handler.setLevel(logging.DEBUG)
+
+        # set up primary log
+        log = logging.getLogger()
+        log.setLevel(logging.DEBUG)
+        log.addHandler(handler)
+        # add the second logger
+        log.addHandler(sysloghandler)
+    except Exception as ex:
+        raise ExportException(sdstatus=Status.ERROR_LOGGING) from ex
+
+
+def _start_service(submission: Archive) -> Status:
+    """
+    Start print or export service.
+    """
+    # Print Routines
+    if submission.command is Command.PRINT:
+        return PrintService(submission).print()
+    elif submission.command is Command.PRINTER_PREFLIGHT:
+        return PrintService(submission).printer_preflight()
+    elif submission.command is Command.PRINTER_TEST:
+        return PrintService(submission).printer_test()
+
+    # Export routines
+    elif submission.command is Command.EXPORT:
+        return ExportService(submission).export()
+    elif submission.command is Command.CHECK_USBS:
+        return ExportService(submission).check_connected_devices()
+    elif submission.command is Command.CHECK_VOLUME:
+        return ExportService(submission).check_disk_format()
```

Comment:
Very nit, but food for thoughts: Maybe this could be encapsulated into `securedrop_export.service.Service`? That would allow `main` to ignore everything about how many services are available of what their commands are. (Prompted by the docstring: `Start print **or** export service.` - emphasis mine.)

---

From there, I would even argue that `Command.START_VM` and the other values of `Command` are actually of different nature (the latter may be `service.Command`s?). By the way, in hindsight, that was already suggested by the fact that we found sufficient to give `Command.START_VM` and empty string value. 

This module could import only the type of commands that holds `Command.START_VM`, while the rest would be abstracted, like `PrintService` and `ExportService` would be.

Reply:
Yes, I thought about this, but I didn't feel like it gains us anything in terms of readability--either we add another file (main -> service dispatch -> print or exportservice) or we parse the command twice (main, parse command and launch either export or print service -> parse command again in appropriate service). 

If our `main` file feels too cluttered later on, I'm happy to pull this out into a separate Service file.